app/scripts/proactive/model/Task.js

```diff
@@ -381,11 +397,110 @@ define(
                 ]
 
                 this.controlFlow = {};
-                this.on("change", function (eventName, event) {
-                    undoManager.save()
+
+                // get the GI of the Task
+                var genericInformation = this.get["Generic Info"];
+
+                this.on("change", function(eventName, error) {
+                  if (eventName) {
+                    if (eventName._changing) {
+                      // Check if Generic Info doc has been changed and generate new link if needed
+                      if (eventName.changed.hasOwnProperty('Generic Info') && eventName.changed["Generic Info"]["Property Value"] != "") {
+
+                        var genericInformation = eventName.changed["Generic Info"];
+                        var hasDocumentation;
+
+                        // Regenerate new documentation url
+                        for (var i in genericInformation) {
+                          if (genericInformation[i]["Property Name"].toLowerCase() === 'task.documentation') {
+                            hasDocumentation = true;
+                            var fileContent = "Documentation for the Job \"" + this.get('Name') + "\" \n" + "\n" + "\n";
+                            var fileContent = fileContent + "Documentation value: " + genericInformation[i]["Property Value"] + "\n";
+                            var linkName = genericInformation[i]["Property Value"];
+                            var documentationValue = genericInformation[i]["Property Value"];
+                            break;
+                          } else {
+                            hasDocumentation = false;
+                          }
+                        }
+
+                        if (hasDocumentation) {
+                          // Set New Value, needed for persistance
+                          this.set({
+                            "Generic Info Documentation": "{\"name\":\"" + linkName + "\",\"url\":\"" + documentationValue + "\"}"
+                          });
+
+                          // Needed to update UI
+                          $('a[name="Generic Info Documentation"]').text(documentationValue);
+                          $('a[name="Generic Info Documentation"]').attr("href", documentationValue);

```

Comment:
I think it would be better with to have a variable that concatenate a generic info variable and the string "Documentation"